File: docs/data-tools/codesnippet/VisualBasic/pass-data-between-forms_2.vb
    Private Sub CustomersDataGridView_DoubleClick() Handles CustomersDataGridView.DoubleClick

        Dim SelectedRowView As Data.DataRowView
        Dim SelectedRow As NorthwindDataSet.CustomersRow

        SelectedRowView = CType(CustomersBindingSource.Current, System.Data.DataRowView)
        SelectedRow = CType(SelectedRowView.Row, NorthwindDataSet.CustomersRow)

        Dim OrdersForm As New Form2
        OrdersForm.LoadOrders(SelectedRow.CustomerID)
        OrdersForm.Show()
    End Sub

File: docs/data-tools/codesnippet/CSharp/pass-data-between-forms_2.cs
        private void customersDataGridView_DoubleClick(object sender, EventArgs e)
        {
            System.Data.DataRowView SelectedRowView;
            NorthwindDataSet.CustomersRow SelectedRow;

            SelectedRowView = (System.Data.DataRowView)customersBindingSource.Current;
            SelectedRow = (NorthwindDataSet.CustomersRow)SelectedRowView.Row;

            Form2 OrdersForm = new Form2();
            OrdersForm.LoadOrders(SelectedRow.CustomerID);
            OrdersForm.Show();
        }